dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,188 @@
+        final CountDownLatch countDownLatch = new CountDownLatch(THREAD_AMOUNT);
+
+        for (int i = 0; i < THREAD_AMOUNT; i++) {
+            final Thread worker = new Thread(
+                    new StatefulTestThread(cyclicBarrier, countDownLatch, scheduler));
+            worker.setName("Thread " + i);
+            worker.start();
+        }
+
+        //wait until all threads are executed to verify the assert
+        countDownLatch.await(15, TimeUnit.SECONDS);
+
+        assertEquals(1, scheduler.getTriggerNames("StatefulTestJob_trigger_group").length);
+    }
+
+
+    /**
+     * <b>Method to test:</b> {@link TriggerBuilder#build()}
+     * <p>
+     * <b>Given scenario:</b> Multiple threads trigger jobs in a stateful job
+     * <p>
+     * <b>Expected result:</b> Only one trigger should be created to handle all the jobs
+     */
+    @Test
+    public void testTriggerBuilder() throws InterruptedException {
+
+        final CyclicBarrier cyclicBarrier = new CyclicBarrier(THREAD_AMOUNT);
+        final CountDownLatch countDownLatch = new CountDownLatch(THREAD_AMOUNT);
+        final DotStatefulTestJob dotStatefulTestJob = new DotStatefulTestJob();
+
+        for (int i = 0; i < THREAD_AMOUNT; i++) {
+            final Thread worker = new Thread(
+                    new StatefulBuilderTestThread(cyclicBarrier, countDownLatch, dotStatefulTestJob));
+            worker.setName("Thread " + i);
+            worker.start();
+        }
+
+        //wait until all threads are executed to verify the assert
+        countDownLatch.await(15, TimeUnit.SECONDS);
+
+        assertTrue(DotStatefulTestJob.getThreadsExecuted() > 1);
+    }
+
+    /**
+     * Inner class with the logic to be executed by each thread in {@link
+     * DotStatefulJobTest#testDotStatefulJobIsReallyStateful()} ()}
+     */
+    private class StatefulTestThread implements Runnable {
+
+        private CyclicBarrier cyclicBarrier;
+        private CountDownLatch countDownLatch;
+        private Scheduler scheduler;
+
+        private StatefulTestThread(final CyclicBarrier cyclicBarrier,
+                final CountDownLatch countDownLatch,
+                final Scheduler scheduler) {
+            this.cyclicBarrier = cyclicBarrier;
+            this.countDownLatch = countDownLatch;
+            this.scheduler = scheduler;
+        }
+
+        @Override
+        public void run() {
+            final String thisThreadName = Thread.currentThread().getName();
+
+            try {
+                final JobDetail jobDetail = getJobDetail(thisThreadName);
+
+                final Trigger trigger = new TriggerBuilder().jobDetail(jobDetail)
+                        .triggerGroupName("StatefulTestJob_trigger_group").build();
+
+                //wait for all threads to schedule jobs concurrently
+                cyclicBarrier.await();
+                scheduler.scheduleJob(jobDetail, trigger);
+                countDownLatch.countDown();
+            } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5971105)